src/Parsers/ParserCreateQuery.h

```diff
@@ -199,6 +200,10 @@ bool IParserColumnDeclaration<NameParser>::parseImpl(Pos & pos, ASTPtr & node, E
         default_specifier = "EPHEMERAL";
         if (!expr_parser.parse(pos, default_expression, expected) && type)
             default_expression = std::make_shared<ASTLiteral>(Field());
+        /// The default expression for EPHEMERAL should be ASTLiteral,
+        /// otherwise it will be crash in formatImpl.
+        if (!startsWith(default_expression->getID(), "Literal"))
```

Comment:
String comparison looks very strange - it is "unstructured" code. That's what you typically do in Perl, not in C++.